docs/project-structure.md

```diff
@@ -0,0 +1,58 @@
+## Development specifications
+
+Proposal regarding the project structure and related tools
+
+### How to run
+
+```
+external-dns --outside-cluster --dnsprovider=aws --source=ingress --source=service
+```
+
+### Vendoring tool 
+
+- glide 
+- alternatives: govendor, godep
+
+### Project structure
+
+```
+./main.go
+./config.go - store configurations, flag parsing
+./controller - main controlling loop
+    controller.go 
+./plan/
+    record.go - dns provider neutral struct for records
+    plan.go - implements the logic for managing records
+./kubernetes/
+    manager.go - provides watching capabilities + clientset
+./dnsprovider/ - dns providers
+    aws.go
+    google.go
+    fake.go 
+    dnsprovider.go - interface
+./source/ - list of sources
+    fake.go
+    ingress.go
+    services.go
+    source.go - interface
+```
+
+### Dependencies 
```

Comment:
Dependencies we'll almost certainly have not listed so far:
```
"github.com/kubernetes/client-go"
"github.com/aws/aws-sdk-go"
"github.com/GoogleCloudPlatform/google-cloud-go"
```

Perhaps I should add a "Cloud Provider" group?

Kubernetes dependency probably stands alone, unless we plan on using some already existing dns provider work.